### ISSUE TITLE
Show memory usage and load time in extension manager

### ISSUE THREAD

**Issue by krassowski** [jupyterlab/jupyterlab#13402]
### Problem

With many extensions installed the performance may slow down. Users have no way to know how much of a performance penalty each extension introduces.

### Proposed Solution

- Measure time for plugin loading and delta in memory usage using `window.performance` functions.
- Discretely show the result in the extension manager (possibly only for plugins exceeding certain load time/size)
- Offer more detail on hover. An additional endpoint could be number of CSS rules contributed.

### Additional context

VSCode shows the activation time:

<details>

![Screenshot from 2022-11-09 17-17-37](https://user-images.githubusercontent.com/5832902/200907690-0a7f1a1e-3f14-4e19-b986-485217f9b9d2.png)

</details>

**Comment by rutviklhase** [comment]
Hey @jweill-aws ! Is anyone working on this? Would love to take this one up!

**Comment by jweill-aws** [comment]
@rutviklhase Nobody's picked this one up yet. Please feel free to submit a pull request if you'd like to implement this!

**Comment by rutviklhase** [comment]
@jweill-aws Awesome! I'll start working on it

**Comment by krassowski** [comment]
> Not necessarily in this PR, but maybe it would also be useful to add these values to the Advanced Plugin Manager, for example in a new column?
> 
> So plugins could be sorted by the time they take to activate, which would help identify potential issues with some third-party extensions.
> 
> <img width="3600" height="2090" alt="image" src="https://github.com/user-attachments/assets/ded01910-7b52-4ac9-9b83-b8903a5c813c" />
> 
> _Originally posted by @jtpio in https://github.com/jupyterlab/jupyterlab/issues/18181#issuecomment-3654103549_

Indeed, we could show it in both the Plugin Manager and the Extension Manager, with the former giving a more granular view and providing frontend-only metrics; the time in Extension Manager could also show server-side loading cost.
